connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1063,6 +1076,132 @@ public int generation() {
         return generation;
     }
 
+    @Override
+    public void restartConnectorAndTasks(
+            RestartRequest request,
+            Callback<ConnectorStateInfo> callback
+    ) {
+        final String connectorName = request.connectorName();
+        addRequest(
+                () -> {
+                    if (checkRebalanceNeeded(callback)) {
+                        return null;
+                    }
+                    if (!configState.connectors().contains(request.connectorName())) {
+                        callback.onCompletion(new NotFoundException("Unknown connector: " + connectorName), null);
+                        return null;
+                    }
+                    if (isLeader()) {
+                        // Write a restart request to the config backing store, to be executed asynchronously in tick()
+                        configBackingStore.putRestartRequest(request);
+                        // Compute and send the response that this was accepted
+                        Optional<RestartPlan> maybePlan = buildRestartPlan(request);
```

Comment:
I'd call it `plan` here.

Reply:
The plan may or may not be present, that's why I was calling it maybePlan

Reply:
We might be too precise here. As we discussed elsewhere, baking the meaning of the type in every variable name might be too verbose. 
It's a plan, and the fact that is optional means that there might be one or it might not. That's how I'd read it. 
The other use of `maybePlan` below is harder to avoid. Feel free to keep it consistent here with what you have below. 
Again, my point is not to bake type meaning in the variable names. I feel this keeps things simpler.

Reply:
@kkonstantine after your latest explanation it makes sense to rename it to keep it simple. Thanks for guiding me. I have renamed it.